### PR TITLE
[2.13.x] DDF-4035: Add Policies and Enable Sec Manager for Standalone Solr

### PR DESCRIPTION
**Pull request by garrettfreibott**
Master port: https://github.com/codice/ddf/pull/3556

#### What does this PR do?
Enables the Pro-Grade security manager on the JVM running the standalone Solr instance.

#### Who is reviewing it? 
@mcalcote 
@ahoffer 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@coyotesqrl 
@stustison

#### How should this be tested?
Run itests and verify that all Solr functionality such and ingest, query, and retrieval still work as expected.

#### What are the relevant tickets?
[DDF-4053](https://codice.atlassian.net/browse/DDF-4035)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/solr-distro/src/main/resources/security/solr-default.policy [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    // Modify application permissions at will.
+    permission java.security.SecurityPermission "getDomainCombiner";
+    permission java.security.SecurityPermission "setPolicy";
+    permission java.security.SecurityPermission "insertProvider";
+    permission java.security.SecurityPermission "removeProvider.*";
+    permission java.security.SecurityPermission "clearProviderProperties.*";
+    permission java.security.SecurityPermission "removeProviderProperty.*";
+
+    // Load classes into any protection domain
+    permission java.lang.RuntimePermission "createClassLoader";
```
Comment (coyotesqrl) [inline distribution/solr-distro/src/main/resources/security/solr-default.policy]: ✏️ There doesn't seem to be any point in denying this permission in this global block only to globally grant it on line 32.

File: distribution/solr-distro/src/main/resources/security/solr-default.policy [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+grant {
+    // Solr folder access permissions
+    permission java.io.FilePermission "..", "read, write, delete";
```
Comment (coyotesqrl) [inline distribution/solr-distro/src/main/resources/security/solr-default.policy]: ❓ Are these going to be relative to this directory or to the directory from which the user launches the application?

Comment (garrettfreibott) [reply distribution/solr-distro/src/main/resources/security/solr-default.policy]: I tested to make sure that it it stays relative to the location of the solr start.jar itself which is at `ddf_home/solr/server/start.jar`, so it gives permission to anything under `ddf_home/solr`

**Comment by garrettfreibott** [comment]
build now

**Pull request merged**